builtin/providers/aws/resource_aws_redshift_cluster.go

```diff
@@ -345,24 +349,40 @@ func resourceAwsRedshiftClusterRead(d *schema.ResourceData, meta interface{}) er
 
 	d.Set("cluster_public_key", rsc.ClusterPublicKey)
 	d.Set("cluster_revision_number", rsc.ClusterRevisionNumber)
+	d.Set("tags", tagsToMapRedshift(rsc.Tags))
 
 	return nil
 }
 
 func resourceAwsRedshiftClusterUpdate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).redshiftconn
+	d.Partial(true)
 
+	arn, tagErr := buildRedshiftARN(d.Id(), meta.(*AWSClient).accountid, meta.(*AWSClient).region)
+	if tagErr != nil {
+		log.Printf("[DEBUG] Error building ARN for Redshift Cluster, not updating Tags for cluster %s", d.Id())
```

Comment:
This is how it is currently done in db_instance as well

https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_db_instance.go#L659

I just used the same layout here